File: problems/215_Kth_Largest_Ele.py

class Solution:
    def findKthLargest(self, nums: list[int], k: int) -> int:
        nums.sort()
        #print(f"Sorted List: {nums}")
        return nums[-k]
        

def main():
    nums_1 = [3,2,1,5,6,4]
    k_1 = 2
    nums_2 = [3,2,3,1,2,4,5,5,6]
    k_2 = 4
    
    leet = Solution()
    print(f"1.Answer of 1stQ:{leet.findKthLargest(nums_1, k_1)}")
    print(f"2.Answer of 2stQ:{leet.findKthLargest(nums_2, k_2)}")

if __name__ == "__main__":
    main()

File: problems/215_Kth_Largest_Ele.c
#include <string.h>
#include <stdlib.h>
#include <stdio.h>

int cmpfunc (const void *a, const void *b) {
    return (*(int *)a - *(int *)b);
}

int findKthLargest(int* nums, int numsSize, int k) {
    int index = 0;

    printf("nums_size: %d \n", numsSize);
    qsort(nums, numsSize, sizeof(int), cmpfunc);

    for (index = 0; index < numsSize; index++) {
        printf("%d ", nums[index]);
    }

    printf("\n");

    return nums[numsSize - k];
}

int main () {
    int nums_1[] = {3,2,1,5,6,4};
    int k_1 = 2;
    
    int nums_2[] = {3,2,3,1,2,4,5,5,6};
    int k_2 = 4;

    int ret = 0;

    ret = findKthLargest(nums_1, sizeof(nums_1)/sizeof(int), k_1);
    printf("1. Answer of 1st-Q: %d \n", ret);

    ret = findKthLargest(nums_2, sizeof(nums_2)/sizeof(int), k_2);
    printf("2. Answer of 2nd-Q: %d \n", ret);

    return 0;
}